src/dowel/simple_outputs.py

```diff
@@ -75,31 +83,38 @@ class TextOutput(FileOutput):
     """Text file output for logger.
 
     :param file_name: The file this output should log to.
+    :param keys_accepted: Regex for which keys this output should accept.
     :param with_timestamp: Whether to log a timestamp before the data.
     """
 
-    def __init__(self, file_name, with_timestamp=True):
-        super().__init__(file_name, 'a')
+    def __init__(self, file_name, keys_accepted=r'^', with_timestamp=True):
+        super().__init__(file_name, keys_accepted=keys_accepted, mode='a')
         self._with_timestamp = with_timestamp
-        self._delimiter = ' | '
+        self.tabular = Tabular()
 
     @property
     def types_accepted(self):
-        """Accept str objects only."""
-        return (str, TabularInput)
+        """Accept str and scalar objects."""
+        return (str, ) + np.ScalarType
 
-    def record(self, data, prefix=''):
+    def record(self, key, value, prefix=''):
         """Log data to text file."""
-        if isinstance(data, str):
-            out = prefix + data
-            if self._with_timestamp:
-                now = datetime.datetime.now(dateutil.tz.tzlocal())
-                timestamp = now.strftime('%Y-%m-%d %H:%M:%S')
-                out = '%s | %s' % (timestamp, out)
-        elif isinstance(data, TabularInput):
-            out = str(data)
-            data.mark_str()
+        if not key:
+            if isinstance(value, str):
+                out = prefix + value
+                if self._with_timestamp:
+                    now = datetime.datetime.now(dateutil.tz.tzlocal())
+                    timestamp = now.strftime('%Y-%m-%d %H:%M:%S')
+                    out = '%s | %s' % (timestamp, out)
+                self._log_file.write(out + '\n')
+            else:
+                raise ValueError('Unacceptable type')
         else:
-            raise ValueError('Unacceptable type.')
+            self.tabular.record(key, value)
 
-        self._log_file.write(out + '\n')
+    def dump(self, step=None):
+        """Flush data to log file."""
+        if not self.tabular.empty:
+            self._log_file.write(str(self.tabular) + '\n')
+            self.tabular.clear()
```

Comment:
again i'm not sure tabular needs to be cleared here. not clearing it avoids an entire class of warnings/bugs